tla/consistency/TraceMultiNodeReads.tla

```diff
@@ -81,13 +86,29 @@ IsRoTxResponseAction ==
     /\ RoTxResponseAction
     /\ Last(history').type = ToTxType[logline.type]
     /\ Last(history').tx = logline.tx
+    /\ Last(history').tx_id = logline.tx_id
 
 IsStatusInvalidResponseAction ==
     /\ IsEvent("StatusInvalidResponseAction")
     /\ StatusInvalidResponseAction
     /\ Last(history').type = ToTxType[logline.type]
     /\ Last(history').status = ToStatus[logline.status]
 
+IsNotEvent ==
+    l' = l
+
+InsertTruncateLedgerAction ==
+    /\ IsNotEvent
+    /\ "view" \in DOMAIN logline
+    /\ logline.view > Len(ledgerBranches)
+    /\ "tx_id" \in DOMAIN logline
+    /\ logline.tx_id[1] > Len(ledgerBranches)
+    /\ TruncateLedgerAction
+
+InsertOtherTxnAction ==
+    /\ IsNotEvent
+    /\ AppendOtherTxnAction
```

Comment:
`InsertOtherTxnAction` should be enabled iff `l \in 1..Len(JsonLog)`.

Reply:
That should be covered by the Termination constraint.